services/sonar/sonar-violations.service.js

```diff
@@ -0,0 +1,146 @@
+'use strict'
+
+const { colorScale } = require('../color-formatters')
+const { metric } = require('../text-formatters')
+const SonarBase = require('./sonar-base')
+const {
+  patternBase,
+  queryParamWithFormatSchema,
+  isLegacyVersion,
+  getLabel,
+} = require('./sonar-helpers')
+
+const violationsColorScale = colorScale(
+  [1, 2, 3, 5],
+  ['brightgreen', 'yellowgreen', 'yellow', 'orange', 'red']
+)
+
+const violationCategoryColorMap = {
+  blocker_violations: 'red',
+  critical_violations: 'orange',
+  major_violations: 'yellow',
+  minor_violations: 'yellowgreen',
+  info_violations: 'green',
+}
+
+module.exports = class SonarViolations extends SonarBase {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'violations' }
+  }
+
+  static renderLongViolationsBadge(violations) {
```

Comment:
This is a new badge, but one I think was worth adding as it provides an option for users that we offer in many other similar badges